### ISSUE TITLE
Extension causes high cpu load

### ISSUE THREAD

**Issue by andreyagra** [bung87/vscode-rails#19]
- Issue Type: `Performance`
- Extension Name: `rails`
- Extension Version: `0.8.5`
- OS Version: `Darwin x64 17.7.0`
- VSCode version: `1.30.1`

:warning: Make sure to **attach** this file from your *home*-directory: `/Users/andreyagra/bung87.rails-unresponsive.cpuprofile.txt` :warning:

Find more details here: https://github.com/Microsoft/vscode/wiki/Explain:-extension-causes-high-cpu-load

**Comment by bung87** [comment]
could you provide the `unresponsive.cpuprofile.txt` file?

**Comment by andreyagra** [comment]
bung87.rails-unresponsive.cpuprofile.txt
[bung87.rails-unresponsive.cpuprofile.txt](https://github.com/bung87/vscode-rails/files/2730162/bung87.rails-unresponsive.cpuprofile.txt)

**Comment by bung87** [comment]
thank you! I can't confirm what cause this problem,the profiler log shows it gets stuck in loading js module,I have not face this problem on my rails project developing.

**Comment by pycckuu** [comment]
I do have the same problem. Could it be caused by solargraph?

**Comment by bung87** [comment]
> I do have the same problem. Could it be caused by solargraph?

have no idea, after installing it ,never activate it , status bar always show `starting the solargraph language server`

**Comment by wuffo** [comment]
Same problem.

[bung87.rails-unresponsive.cpuprofile.txt](https://github.com/bung87/vscode-rails/files/3087094/bung87.rails-unresponsive.cpuprofile.txt)

**Comment by leifcr** [comment]
yes. it's solargraph. I recommend removing the dependency, so solargraph can be optional.

By watching 'top' on the console, I can see the ruby process for solargraph constantly running. I guess ruby language server for rails projects isn't ready for prime-time.

**Comment by bung87** [comment]
thank you guys! I'm still not sure what reason is, and I'm working a rails project that I enabled solargraph and vscode-rails never happened the problem you faced,  and I upgrade a version that highly relys on solargraph that make this extension more simpler.

@leifcr think solargraph constantly running is not the problem as it's a ruby language server, if there is it should be the solargraph highly consuming cpu time, but I can't see it from the log.

**Comment by bung87** [comment]
does this problem still existed?  
I starting manage all information I have
1. developers report high cpu load begin at v0.8.5 make solargraph as extension dependency starts v0.9.0
https://github.com/bung87/vscode-rails/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+Extension+causes+high+cpu+load

so problem exists in one extension side, or both side, even it 's not as vscode-rails extension dependency, as they register same providers.

also see https://github.com/castwide/vscode-solargraph/issues/116 Extension causes high cpu load
seems the author make it  more efficient.

2. this project performance related parts

    1. as rails project has standard directory struct and file name, when searching file using 
    `vscode.workspace.findFiles` it always using most near sub path.  
    2. after find file it scan line by line to get symbol info

3. there may have a bug after let solargraph as denpendency, I also commented some lines.

https://github.com/bung87/vscode-rails/blob/97e3b6184ea114d245eaf53f314d268eaf1c1f46/src/rails_definition.ts#L485

it may introduce this error https://github.com/bung87/vscode-rails/issues/45

4. about not use solargraph as extension dependency

it's also sound make sense , leave language level extension to end user choises
the bad side is without any ruby language server, the end user must be a ruby pro.

**Comment by zarembas** [comment]
Might be a Ruby version issue. It pops up all the time on a Ruby 2.3.3 project I'm working on.
